### PR TITLE
feature: add get assignment controller

### PR DESCRIPTION
**Pull request by LBokan**
### Description

Added a controller for getting user's assignments and photos

### Type of change

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [ ] Configuration
- [x] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

Add screenshots of the changed components here.

#### Screenshots/Video after change

Add screenshots with the changes applied here.

### CODE REVIEW

File: backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class AssignmentMapper {
+  public static AssignmentEntity mapToAssignment( AssignmentRequestEntity assignmentRequest ) {
+    return AssignmentEntity
+        .builder()
+        .dateOfIncident( assignmentRequest.getDateOfIncident() )
+        .build();
+  }
```
Comment (rmaskalenka) [inline backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java:20]: Assignment Entity has more than 1 field. Is this expected?

File: backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  public static AssignmentResponseEntity mapToAssignmentResponse( AssignmentEntity assignment ) {
```
Comment (rmaskalenka) [inline backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java:22]: Please don't use static methods for mapper. By using static methods for such operations you will have high coupling between your components. This is also much less testable as classes that will use such menhods will not have any easy ways to mock their behavior

File: backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import com.exadel.carinsurance.model.request.AssignmentRequestEntity;
+import com.exadel.carinsurance.model.response.AssignmentResponseEntity;
+import com.exadel.carinsurance.model.response.ContactInfoResponseEntity;
+import com.exadel.carinsurance.model.response.DirectionOfImpactResponseEntity;
+import com.exadel.carinsurance.model.response.VehicleConditionInfoResponseEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AssignmentMapper {
```
Comment (rmaskalenka) [inline backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java:14]: It is good to have a single interface for as much mappers as possible. Please create a common generic interface for your mappers.

File: backend/src/main/java/com/exadel/carinsurance/service/implementation/AssignmentService.java [hunk 4]
```diff
@@ -143,28 +153,56 @@ public ResponseEntity createAssignment(
...
         .ok()
         .body( assignmentEntityFromDB.getAssignmentId() );
   }
+
+  @Override
+  public ResponseEntity getAssignments() {
+    UserEntity user = ( UserEntity ) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+    List<AssignmentResponseEntity> assignmentsResponse = new ArrayList<>();
+
+    for ( AssignmentEntity assignment : user.getAssignments() ) {
```
Comment (rmaskalenka) [inline backend/src/main/java/com/exadel/carinsurance/service/implementation/AssignmentService.java:180]: umm why does user has assignments inside?
are you suggesting to extract all assignments on user information receival?
Please don't. It is better to have assignments list as separate Repository call. Imagine having 10+ users with 100+ assignments for each, seccurity User can be a cached entity so you will store 1000+ records in memory for your application for unknown period of time.

File: backend/src/main/java/com/exadel/carinsurance/service/implementation/PhotosService.java [hunk 5]
```diff
@@ -55,4 +63,50 @@ public PhotosResponseEntity savePhotos( Long userId, Long assignmentId, List<Mul
         .namesOfPhotosOfImpact( String.join( ";", namesOfPhotosOfImpactList ) )
         .build();
   }
+
+  @Override
+  public ResponseEntity getPhoto(
+      @PathVariable( "userId" ) Long userId,
+      @PathVariable( "assignmentId" ) Long assignmentId,
+      @PathVariable( "photoName" ) String photoName
```
Comment (rmaskalenka) [inline backend/src/main/java/com/exadel/carinsurance/service/implementation/PhotosService.java:71]: why does a service uses a controller level annotations? What is the point in those?

Comment (LBokan) [reply backend/src/main/java/com/exadel/carinsurance/service/implementation/PhotosService.java:71]: Removed it. Done

File: backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java (hunk 3 above)
Comment (LBokan) [reply backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java:14]: Created it. Done

File: backend/src/main/java/com/exadel/carinsurance/service/implementation/AssignmentService.java (hunk 4 above)
Comment (LBokan) [reply backend/src/main/java/com/exadel/carinsurance/service/implementation/AssignmentService.java:180]: Done: Removed assignments list from UserEntity, made request to DB to get assignments list

File: backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+import com.exadel.carinsurance.model.response.DirectionOfImpactResponseEntity;
+import com.exadel.carinsurance.model.response.VehicleConditionInfoResponseEntity;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class AssignmentMapper implements IMapper<AssignmentEntity, AssignmentRequestEntity, AssignmentResponseEntity> {
```
Comment (rmaskalenka) [inline backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java:17]: it is better to have mappers for 2 entries not 3.
Think of it as a single responsibility mapper which purpose is to convert object from <source> to <target> and vice versa

**Review by rmaskalenka — APPROVED** [review]

File: backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java (hunk 6 above)
Comment (LBokan) [reply backend/src/main/java/com/exadel/carinsurance/mapper/AssignmentMapper.java:17]: Done

**Pull request merged**